From df9987b1e1f030065115fe3fc66496a2fd9dfe0a Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Date: Mon, 11 Jan 2021 14:14:23 +0200
Subject: [PATCH 42/78] can: llce: Don't report EPROBE_DEFER errors

Avoid errors when the driver is loaded before LLCE mailbox.

Issue: ALB-6286
Signed-off-by: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
---
 drivers/net/can/llce_can.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/net/can/llce_can.c b/drivers/net/can/llce_can.c
index 4dc2ea0ee4dd..b40e070c8c4b 100644
--- a/drivers/net/can/llce_can.c
+++ b/drivers/net/can/llce_can.c
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: GPL-2.0+ OR BSD-3-Clause
-/* Copyright 2020 NXP */
+/* Copyright 2020-2021 NXP */
 #include <linux/can/dev.h>
 #include <linux/clk.h>
 #include <linux/ctype.h>
@@ -642,6 +642,8 @@ static void *get_netdev_name(struct device *dev)
 
 static int init_llce_chans(struct llce_can *llce, struct device *dev)
 {
+	long ret = 0;
+
 	llce->config_client.dev = dev;
 	llce->config_client.tx_block = true;
 	llce->config_client.rx_callback = config_rx_callback;
@@ -656,11 +658,13 @@ static int init_llce_chans(struct llce_can *llce, struct device *dev)
 	llce->config = mbox_request_channel_byname(&llce->config_client,
 						   "config");
 	if (IS_ERR(llce->config)) {
-		dev_err(dev, "Failed to get config mailbox\n");
-		return PTR_ERR(llce->config);
+		ret = PTR_ERR(llce->config);
+		if (ret != -EPROBE_DEFER)
+			dev_err(dev, "Failed to get config mailbox\n");
+		return ret;
 	}
 
-	return 0;
+	return ret;
 }
 
 static int llce_init_can_priv(struct llce_can *llce, struct device *dev)
-- 
2.25.1

